pkg/k8s/clinet_options.go

```diff
@@ -0,0 +1,43 @@
+package k8s
+
+import "k8s.io/client-go/rest"
+
+type ClientOptions struct {
+	clusterDomain string
+	clusterHost   string
+	authToken     string
+	cert          string
+	clusterConfig *rest.Config
+}
+
+type Option func(*ClientOptions)
```

Comment:
Didn't we have this discussion with the KNUU options? It's less code to maintain to just have a `New` and `NewWithOptions` function that accepts the options params as opposed to having a unique function call for each option. Then all the input validation can be contained in one place too instead of across all the individual option functions.